### ISSUE TITLE
[REVM-15] Upgrade to ethers v6

### ISSUE THREAD

**Issue by joewagner** [tablelandnetwork/evm-tableland#337]
Now that Ethers v6 has been released we should start to work towards using it here so that the SDK can upgrade to v6.
There will be some complexity in upgrading. This package has ethers as a dev dependency, and some of the other dev dependencies mark ethers as a peer dependency.  Those peer dependencies all currently have ethers v5 tagged.  Looking at the github repos for these packages makes it seem like they are all going to have versions that use v6 coming out soon.
We should be ready for all those upgrades here.

related: 
https://github.com/NomicFoundation/hardhat/issues/3639
https://github.com/tablelandnetwork/js-tableland/issues/409

<sub>[REVM-15](https://linear.app/tableland/issue/REVM-15/upgrade-to-ethers-v6)</sub>

**Comment by sanderpick** [comment]
oop, the issue is over in Linear now. I had synced / unsynced this repo multiple times when testing the sync tool. so ended up just unsyncing issues. we could add the public tag on linear side if we wanted to make a new issue show up here.

**Comment by joewagner** [comment]
> oop, the issue is over in Linear now. I had synced / unsynced this repo multiple times when testing the sync tool. so ended up just unsyncing issues. we could add the public tag on linear side if we wanted to make a new issue show up here.

That makes sense. I think it would be good to have this as public since this is blocked by a public PR in hardhat and it's nice for folks to be able to see all of that in github
